## Stacks Pipeline Batch 2 - meristics




This notebook will keep track of and visualize the number of reads per individual, number of loci per individual, number of loci in the catalog or reference genome, etc. 

<br>
### Raw read files
**Fastqc of raw read files, per lane: **

[Evernote notebook]() contains HTML files. 

<br>
### Post process_radtags

**Number of reads per individual, post-process radtags**

![graph](https://github.com/mfisher5/PCod-US-repo/blob/master/analyses/stacks_pipeline_visuals/batch2_reads.png?raw=true)


<br>
**Fastqc of some sample fastq files: **

[Evernote notebook]() contains HTML files. 


<br>
### From the FIRST round of stacks
### Post ustacks

<br>
**Total number of tags per individual**
![png tags](https://github.com/mfisher5/PCod-US-repo/blob/master/analyses/stacks_pipeline_visuals/batch2_tags.png?raw=true)




### cstacks

Number of loci in the catalog: 

In [1]:
pwd

u'/mnt/hgfs/Pacific cod/DataAnalysis/PCod-US-repo/notebooks'

In [2]:
cd ../stacks_b2

/mnt/hgfs/Pacific cod/DataAnalysis/PCod-US-repo/stacks_b2


In [3]:
!gzip -d batch_2.catalog.tags.tsv.gz

gzip: batch_2.catalog.tags.tsv.gz: No such file or directory


I must have unzipped that file already...

In [4]:
!grep "consensus" batch_2.catalog.tags.tsv | wc -l

459504


There are **459,504 loci in the r1 cstacks catalog**

*compared to 74,721 loci in the batch 1 cstacks catalog*



<br>
### sstacks


<br>
### populations

Number of loci output in the raw `populations` file:

In [5]:
cd ../scripts

/mnt/hgfs/Pacific cod/DataAnalysis/PCod-US-repo/scripts


In [6]:
!head UniqueLocusCount.py

In [8]:
!python UniqueLocusCount.py ../stacks_b2/batch_2.genepop

Stacks version 1.44; Genepop version 4.1.3; May 24, 2017

With the ORIGINAL script:
The number of unique loci in  ../stacks_b2/batch_2.genepop is:  20460


With the NEW script, used in the pipeline:
Stacks version 1.44; Genepop version 4.1.3; May 24, 2017

Done reading loci

The number of unique loci in  ../stacks_b2/batch_2.genepop is:  20460


<br>

### Reference Genome

<br>
**20234 loci** after BOWTIE filtering


**19509 loci** after BOWTIE and BLAST filtering

<br>
<br>

### From the SECOND round of stacks
### Post pstacks




<br>
### Post cstacks (r2)




In [9]:
cd ../stacks_b2_wgenome

/mnt/hgfs/Pacific cod/DataAnalysis/PCod-US-repo/stacks_b2_wgenome


In [10]:
!grep "consensus" batch_2.catalog.tags.tsv | wc -l

19506


There are **19,506 loci ** in the cstacks catalog, r2

<br>


### populations

In [11]:
cd ../scripts

/mnt/hgfs/Pacific cod/DataAnalysis/PCod-US-repo/scripts


In [13]:
!python UniqueLocusCount.py ../stacks_b2_wgenome/batch_2.genepop

Stacks version 1.44; Genepop version 4.1.3; May 25, 2017

With the ORIGINAL script:
The number of unique loci in  ../stacks_b2_wgenome/batch_2.genepop is:  10729


With the NEW script, used in the pipeline:
Stacks version 1.44; Genepop version 4.1.3; May 25, 2017

Done reading loci

The number of unique loci in  ../stacks_b2_wgenome/batch_2.genepop is:  10729


There are **10,729** loci left after r2 `populations`. 

*compared to 9,717 in batch 1*


<br>
### post-stacks filtering

**Marine's filtering / genotype correcting script: **

Just by opening Excel, it looks like there are only **5,864 loci** left. 

<br>
** Without filtering out individuals with >20% missing data: **

*MAF filtering:* 4528 loci

    compared to 4,048 loci in batch 1

<br>
*Missing Data filtering (loci with >20% missing were filtered out):* 2945

    comapred to 2,098 loci in batch 1


** I ended up with 2,945 loci**

*This is approximately 800 more loci than retained in batch 1. *